src/NServiceBus.AcceptanceTesting/Support/EnforceSubscriptionPublisherMetadataBehavior.cs

```diff
@@ -0,0 +1,30 @@
+namespace NServiceBus.AcceptanceTesting.Support;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Pipeline;
+
+public class EnforceSubscriptionPublisherMetadataBehavior(string endpointName, PublisherMetadata publisherMetadata) : IBehavior<ISubscribeContext, ISubscribeContext>
+{
+    readonly HashSet<Type> eventTypeMap = publisherMetadata.Publishers.SelectMany(publisher => publisher.Events).ToHashSet();
```

Comment:
Is it worth it to use `[FrozenSet<T>](https://learn.microsoft.com/en-us/dotnet/api/system.collections.frozen.frozenset-1?view=net-8.0)` as this is read only data.

Reply:
No. My experience has shown they have a significant initialization cost that can only be amortized when you read thousands and thousands of times. This set is unique per run and endpoint and will only be queried a few times